apps/opencs/view/world/scriptsubview.cpp

```diff
@@ -210,18 +212,28 @@ void CSVWorld::ScriptSubView::useHint (const std::string& hint)
     unsigned line = 0, column = 0;
     char c;
     std::istringstream stream (hint.c_str()+1);
-    switch(hint[0]){
+    switch(hint[0])
+    {
         case 'R':
         case 'r':
         {
             QModelIndex index = mModel->getModelIndex (getUniversalId().getId(), mColumn);
             QString source = mModel->data (index).toString();
+            unsigned stringSize = source.length();
             unsigned pos, dummy;
             if (!(stream >> c >> dummy >> pos) )
                 return;
 
-            for (unsigned i = 0; i <= pos; ++i){
-                if (source[i] == '\n'){
+            if (pos >= stringSize)
```

Comment:
This looks like an off-by-one error to me, since it is possible to position the cursor past the last character of the script text.

Reply:
Note that loop uses a <= statement instead of <.
For example, with pos = 1000 stringSize should be > 1000 to handle 1000-th character in string.
So the situation pos = stringSize is invalid too.

Can you explain what do you mean?

Reply:
Lets make it simple. Assume a single line file with a single character in it. The string has the size 1. Legal cursor positions then would be 0 (on the character) and 1 (after the character). Note that s[s.length()]==0, so there is no need for a special case. At least that is the case for std::string. I just noticed that we are using a bloody QString here. I am 99% sure that it behaves in the same way as std::string. But it wouldn't hurt to check the documentation.

Reply:
> int QString::length() const
> Returns the number of characters in this string. Equivalent to size().
https://doc.qt.io/qt-5/qstring.html#length

> int QString::size() const
> Returns the number of characters in this string.
> 
> The last character in the string is at position size() - 1.
> 

Example:
```
QString str = "World";
int n = str.size();         // n == 5
str.data()[0];              // returns 'W'
str.data()[4];              // returns 'd'
```

Reply:
Interesting. I tried to use the `str.data()[1000000];`, but it just returns zeroes.
So in theory possible error should cause only redundant iterations, and we can just print warning here.

Reply:
So what did we decide to do here?

Reply:
Increased limit by 1 for sure.